### PR TITLE
Say Hello World

### LINKED ISSUE #1124: [Demand] "Post first encrypted post" workflow in Initialization
## Metadata

Field | Value
--- | ---
Previous Stage | #407, #600, #409, #1026 
Figma Link | https://www.figma.com/file/4DmpRxZk4jDeqrfA5xrPCp/Dashboard-v3?node-id=1105%3A0
Expected UX prototype due | 2020-W??-5
Actual UX prototype release | -
Expected engineering due | 2020-W??-5
Corresponding Pull Request | -
Completion Version | -

## Introduction

One more series of steps to be appended for Initialization.

## Documentation

https://confluence.dimension.chat/x/PIAf

Section "Workflow 3: Create the First Encrypted Post".

## Notes

- None.

### PR DESCRIPTION
**Pull request by guanbinrui**
- [x] Implement the UI of `<SayHelloWorld />` step
- [x] Implement `taskOpenComposeBox` in Twitter
- [x] Implement `taskOpenComposeBox` in Facebook
- [x] Implement `taskGotoNewsFeedPage` in Twitter
- [x] Implement `taskGotoNewsFeedPage` in Facebook
- [x] Refactor `currentImmersiveSetupStatus` setting
- [ ] <s>Remove `<PasteIntoBio />` step</s>
- [ ] <s>Implement Minimized UI in mobile</s>
- [ ] <s>Implement the UI of  `<SetupDone />` step</s>

close #1124

say-hello

### CODE REVIEW

File: src/_locales/zh/messages.json [hunk 1]
```diff
@@ -18,15 +18,16 @@
...
     "additional_post_box__encrypted_failed": "請將以下内容拷貝至發佈框。",
     "additional_post_box__encrypted_post_pre": "來 #Maskbook 解密這個訊息！ {{encrypted}}",
     "additional_post_box__steganography_post_failed": "接下來，Maskbook 會將含有加密訊息的圖片保存到您的設備上。請將這張保存了的圖片附加到您的帖子上，這樣別人才能解密其中的訊息。",
     "additional_post_box__steganography_post_pre": "此圖片僅分享給我的好友。請安裝 Maskbook 以解讀圖片。 ([I:b]) {{random}}",
-    "advanced": "高級……",
+    "advanced": "高級…",
     "advanced_options": "高級",
     "database_management": "資料庫管理",
     "automation_pasted_into_bio_box": "[Maskbook] 我們幫助您在簡介中填入了公鑰，請您審閱並發布它！",
+    "automation_request_click_post_button": "Please click the “Post” button to open the compose dialog.",
```
Comment (Jack-Works) [inline src/_locales/zh/messages.json:30]: 👀

File: src/_locales/zh/messages.json [hunk 2]
```diff
@@ -171,7 +174,13 @@
     "immersive_setup_add_bio_title": "粘貼进您的個人簡介",
     "immersive_setup_add_bio_text": "在您的個人簡介中添加以下文本以訪問高級功能。 請不要刪除或更改它。",
     "immersive_setup_paste_into_bio_auto": "幫我加上它",
-    "immersive_setup_paste_into_bio_failed": "失敗了……請手動將其添加到您的個人簡介中",
+    "immersive_setup_paste_into_bio_failed": "失敗了… 請手動將其添加到您的個人簡介中",
+    "immersive_setup_create_post_auto": "現在創建",
+    "immersive_setup_create_post_failed": "失敗了… 請手動將其添加到編輯框中",
+    "immersive_setup_say_hello_title": "說「Hello World」",
```
Comment (Jack-Works) [inline src/_locales/zh/messages.json:180]: 👀......`給大家打個招呼！`

File: src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx [hunk 3]
```diff
@@ -433,29 +516,98 @@ export function SetupGuide(props: SetupGuideProps) {
...
+                    username,
+                    persona: persona.toText(),
+                } as ImmersiveSetupCrossContextStatus)
+                const connecting = new ProfileIdentifier(ui.networkIdentifier, username)
+                await Services.Identity.attachProfile(connecting, persona, { connectionConfirmState: 'confirmed' })
+                ui.taskGotoProfilePage(connecting)
+                setStep(SetupGuideStep.PasteIntoBio)
+                break
+            case SetupGuideStep.PasteIntoBio:
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
```
Comment (Jack-Works) [inline src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx:541]: Use `json-stable-stringify` in this case or a unintented infinity loop might comes in.

File: src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx [hunk 4]
```diff
@@ -433,29 +516,98 @@ export function SetupGuide(props: SetupGuideProps) {
...
-            persona: persona.toText(),
-        } as ImmersiveSetupCrossContextStatus)
-        const connected = new ProfileIdentifier(ui.networkIdentifier, username_)
-        await Services.Identity.detachProfile(connected)
-        setStep(SetupGuideStep.FindUsername)
+    const onBack = async () => {
+        switch (step) {
+            case SetupGuideStep.PasteIntoBio:
+                const username_ = getUsername()
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
```
Comment (Jack-Works) [inline src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx:558]: json-stable-stringify

File: src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx [hunk 5]
```diff
@@ -433,29 +516,98 @@ export function SetupGuide(props: SetupGuideProps) {
...
+                    status: SetupGuideStep.FindUsername,
+                    username: '', // ensure staying find-username page
+                    persona: persona.toText(),
+                } as ImmersiveSetupCrossContextStatus)
+                const connected = new ProfileIdentifier(ui.networkIdentifier, username_)
+                await Services.Identity.detachProfile(connected)
+                setStep(SetupGuideStep.FindUsername)
+                break
+            case SetupGuideStep.SayHelloWorld:
+                currentImmersiveSetupStatus[ui.networkIdentifier].value = JSON.stringify({
```
Comment (Jack-Works) [inline src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx:568]: json-stable-stringify

File: src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx [hunk 6]
```diff
@@ -433,29 +516,98 @@ export function SetupGuide(props: SetupGuideProps) {
...
+                ui.taskGotoProfilePage(connecting)
+                setStep(SetupGuideStep.PasteIntoBio)
+                break
+        }
+    }
+    const onPaste = async () => {
+        try {
+            await navigator.clipboard.writeText(provePost)
+            ui.taskPasteIntoBio(provePost)
+            setPastedStatus(true)
```
Comment (Jack-Works) [inline src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx:582]: If you need the paste result, maybe you want to use [useCopyToClipboard](https://github.com/streamich/react-use/blob/master/docs/useCopyToClipboard.md) in react-use libaray, it's cleaner.

File: src/components/Welcomes/Banner.tsx [hunk 7]
```diff
@@ -27,46 +26,44 @@ interface BannerUIProps
...
-    },
-}))
+const useStyles = makeStyles((theme: Theme) => {
+    const network = getActivatedUI()?.networkIdentifier
+    return {
+        root: {
+            border: `1px solid ${theme.palette.divider}`,
+            borderRadius: 4,
+            marginBottom: 10,
+            ...(network === 'twitter.com'
```
Comment (Jack-Works) [inline src/components/Welcomes/Banner.tsx:36]: 👀👀👀

File: src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx (hunk 6 above)
Comment (guanbinrui) [reply src/components/InjectedComponents/ImmersiveGuide/SetupGuide.tsx:582]: Not necessary here.

File: src/components/Welcomes/Banner.tsx (hunk 7 above)
Comment (guanbinrui) [reply src/components/Welcomes/Banner.tsx:36]: It's cleaner than using an external theme defined in other places.

File: e2e/commands/dashboard.ts [hunk 8]
```diff
@@ -76,6 +76,9 @@ export async function reset(page: Page) {
             } else {
                 break
             }
+
+            // waiting for personas list to be updated
+            await page.waitFor(500)
```
Comment (septs) [inline e2e/commands/dashboard.ts:81]: loop check?

```suggestion
            while (true) {
                ... check element ...
                await page.waitFor(100)
            }
```

**Comment by Tedko** [comment]
👍🏻👍🏻

**Pull request merged**